Makefile

```
@@ -366,29 +366,36 @@ release-compress:
 	fi
 	cd $(DIST)/release/; for file in `find . -type f -name "*"`; do echo "compressing $${file}" && gxz -k -9 $${file}; done;
 
-.PHONY: js
-js:
-	@if ([ ! -d "$(PWD)/node_modules" ]); then \
-		echo "node_modules directory is absent, please run 'npm install' first"; \
+npm-check:
+	@hash npm > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
+		echo "Please install Node.js 8.x or greater with npm"; \
 		exit 1; \
 	fi;
 	@hash npx > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
-		echo "Please install npm version 5.2+"; \
+		echo "Please install Node.js 8.x or greater with npm"; \
 		exit 1; \
 	fi;
+
+.PHONY: npm
+npm:
```

```suggestion
npm: npm-check
```
Instead of calling make directly in these you can add a dependency on another target like this so it runs that target first. Same for npm-update and css (see the 'release' target for another example of this)

---

Can do, tought I also kind like those explicit calls and you can do something like

``` make
$(MAKE) target
do something
$(MAKE) target
````

I've been avoiding make dependencies so far because of bad experience with stale files but I guess there's no risk if all targets are PHONY.